service/slice_config_webhook_validation_test.go

```diff
@@ -1420,7 +1426,135 @@ func ValidateMaxClusterForParticipatingCluster(t *testing.T) {
+			Namespace: namespace,
+		}
+		arg.Spec = controllerv1alpha1.VpnKeyRotationSpec{
+			SliceName:               name,
+			CertificateCreationTime: metav1.Now(),
+			CertificateExpiryTime:   metav1.Time{expiry},
+		}
+	}).Once()
+	oldSliceConfig := controllerv1alpha1.SliceConfig{}
+	err := validateCertsRotationInterval(ctx, sliceConfig, &oldSliceConfig)
+	require.Nil(t, err)
+}
+
+func TestValidateCertsRotationInterval_Negative(t *testing.T) {
+	name := "slice_config"
+	namespace := "randomNamespace"
+	clientMock, sliceConfig, ctx := setupSliceConfigWebhookValidationTest(name, namespace)
+	// RenewBefore is 1 hour after, decline
+	sliceConfig.Spec.RenewBefore = time.Now().Add(time.Hour * 1)
+	expiry := metav1.Now().Add(30)
+	clientMock.On("Get", ctx, mock.Anything, mock.Anything).Return(nil).Run(func(args mock.Arguments) {
+		arg := args.Get(2).(*controllerv1alpha1.VpnKeyRotation)
+		arg.ObjectMeta = metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		}
+		arg.Spec = controllerv1alpha1.VpnKeyRotationSpec{
+			SliceName:               name,
+			CertificateCreationTime: metav1.Now(),
+			CertificateExpiryTime:   metav1.Time{expiry},
+		}
+	}).Once()
+	oldSliceConfig := controllerv1alpha1.SliceConfig{}
+	err := validateCertsRotationInterval(ctx, sliceConfig, &oldSliceConfig)
+	require.NotNil(t, err)
+}
+
+func TestValidateCertsRotationInterval_NegativeClusterStatus(t *testing.T) {
+	name := "slice_config"
+	namespace := "randomNamespace"
+	clientMock, sliceConfig, ctx := setupSliceConfigWebhookValidationTest(name, namespace)
+
+	sliceConfig.Spec.RenewBefore = time.Now()
+	expiry := metav1.Now().Add(30)
+	clientMock.On("Get", ctx, mock.Anything, mock.Anything).Return(nil).Run(func(args mock.Arguments) {
+		arg := args.Get(2).(*controllerv1alpha1.VpnKeyRotation)
+		arg.ObjectMeta = metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		}
+		arg.Spec = controllerv1alpha1.VpnKeyRotationSpec{
+			SliceName:               name,
+			CertificateCreationTime: metav1.Now(),
+			CertificateExpiryTime:   metav1.Time{expiry},
+			ClusterGatewayMapping: map[string][]string{
+				"cluster-1": {"gateway-1"},
+				"cluster-2": {"gateway-2"},
+			},
+		}
+		arg.Status = controllerv1alpha1.VpnKeyRotationStatus{
+			CurrentRotationState: map[string]controllerv1alpha1.StatusOfKeyRotation{
+
+				"gateway-1": controllerv1alpha1.StatusOfKeyRotation{
+					Status:               controllerv1alpha1.Complete,
+					LastUpdatedTimestamp: metav1.Now(),
+				},
+				"gateway-2": controllerv1alpha1.StatusOfKeyRotation{
+					Status:               controllerv1alpha1.InProgress,
+					LastUpdatedTimestamp: metav1.Now(),
+				},
+			},
+		}
+	}).Once()
+	oldSliceConfig := controllerv1alpha1.SliceConfig{}
+	err := validateCertsRotationInterval(ctx, sliceConfig, &oldSliceConfig)
+	require.NotNil(t, err)
+	require.Equal(t, err.Type, field.ErrorTypeForbidden)
+}
+
+func TestValidateCertsRotationInterval_PossitiveClusterStatus(t *testing.T) {
```

Comment:
spell check for TestValidateCertsRotationInterval_PossitiveClusterStatus positive